src/components/TodoList/index.tsx

```diff
@@ -0,0 +1,156 @@
+        <div
+          key={todo.id}
+          className={`group relative flex cursor-pointer items-center justify-between rounded-lg p-1 transition-all duration-200 ${hoverBgColor}`}
+          onClick={() => onTodoClick(todo)}
+          role="button"
+          tabIndex={0}
+          onKeyDown={() => onTodoClick(todo)}
+        >
+          <div className="flex w-full items-start space-x-2 overflow-hidden">
+            <button
+              type="button"
+              className="flex h-6 w-6 flex-shrink-0 items-center justify-center p-0.5"
+              onClick={(e) => {
+                e.stopPropagation();
+                onToggleDone(todo.id);
+              }}
+              aria-label={todo.done ? 'Mark as incomplete' : 'Mark as complete'}
+            >
+              {todo.done ? (
+                <ActiveBlue width={24} height={24} />
+              ) : (
+                <Inactive width={24} height={24} />
+              )}
+            </button>
+            <div className="min-w-0 flex-1 overflow-hidden">
+              <div
+                className={`pt-0.5 text-sm font-normal text-slate-800 ${todo.done ? 'line-through' : ''} truncate whitespace-nowrap`}
+              >
+                {todo.title}
+              </div>
+              {showGoals && todo.goal && (
+                <div className="mt-1 flex items-center overflow-hidden">
+                  <div className="flex h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-slate-50">
+                    <GoalIcon width={24} height={24} />
+                  </div>
+                  <div className="ml-2 truncate text-sm font-normal text-slate-700">
+                    {todo.goal.title}
+                  </div>
+                </div>
+              )}
+            </div>
+          </div>
+          {showIcons && (
+            <div className="flex flex-shrink-0 items-center space-x-2">
+              {todo.fileUrl && (
+                <Link
+                  to={todo.fileUrl}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="hidden h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-slate-50 tablet:flex"
+                  onClick={(e) => e.stopPropagation()}
+                  aria-label="Open file"
+                >
+                  <FileIcon width={24} height={24} />
+                </Link>
+              )}
+              {todo.linkUrl && (
+                <Link
+                  to={todo.linkUrl}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="hidden h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-slate-50 tablet:flex"
+                  onClick={(e) => e.stopPropagation()}
+                  aria-label="Open link"
+                >
+                  <LinkIcon width={24} height={24} />
+                </Link>
+              )}
+              {todo.noteId || todo.noteId === 0 ? (
```

Comment:
스웨거 보면 noteId가 필수고 nullable로 돼있어서 null인지만 검사하면 더 좋지 않을까요?
<img width="271" alt="image" src="https://github.com/user-attachments/assets/5eedc56b-abe8-45d8-befa-9312f47bf23b">

```suggestion
              {todo.noteId !== null ? (
```

Reply:
오 감사합니다!!✨ 수정했습니당